packages/synapse-react-client/src/components/SynapseTable/TablePagination.tsx

```diff
@@ -22,15 +34,16 @@ export const TablePagination = () => {
 
   // PORTALS-2259: Special case.  If we're on the first page,
   // and the total query count is less than the min page size, then do not show pagination UI.
-  if (currentPage == 1 && queryCount && queryCount < 10) {
+  // Also hide pagination if the query count is unavailable.
+  if ((currentPage == 1 && queryCount && queryCount < 10) || !queryCount) {
     return <></>
   }
```

Comment:
I think 10 here should be pageSizeOptionsBasedOnData[0], right?
```suggestion
  const smallestPageSizeOption = pageSizeOptionsBasedOnData[0]
  if ((currentPage == 1 && queryCount && queryCount < smallestPageSizeOption) || !queryCount) {
    return <></>
  }
```

Reply:
This is an interesting case!  I think we can hide the pagination as soon as all of the results are shown in a single page

Reply:
I think that makes this unusual sequence possible:

1. Receive 40 results, 25 results shown per page
2. Change the results shown per page to 40
3. Pagination is now hidden and you can't change it to 10 or back to 25

Reply:
This is true, so it works like the Show More button (expand to all results, no way to contract).  It's kind of strange, but I think the UX is ok.  If you disagree, let's talk about the scenario with design.  (How important is it to hide pagination when the results are less than the page size?)